### ISSUE TITLE
Overriding variables with role parameters doesn't work when vars_files path contains a templated var

### ISSUE THREAD

**Issue by natefoo** [ansible/ansible#8903]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.7.1
ansible 1.5.4
##### Environment:

CentOS 6.5
Mac OS X 10.9.4
##### Summary:

It is impossible to override an inventory, host_vars, or group_vars variable with a role parameter if there is a templated value in a `vars_file[s]` path in the play, such that:
1. The play includes a `vars_file[s]` that includes at least one var (it does not matter if the var is used anywhere).
2. The problematic var (in this case, `foo`), is defined in a matching `group_vars` or `host_vars` file or as an inventory variable.
3. A var in a `group_vars` or `host_vars` file or inventory variable (in this case, `subdir`) is templated into the path to the `vars_file[s]` in 1.

The key is the templating in `vars_files`. If I remove `{{ subdir }}` and just use a string, `foo` will contain `bar` on the second execution of the `test` role. Other things work as well, such as defining `foo` in the play's `vars:` instead of in the inventory/host_vars/group_vars.

I can work around this various ways, but it's such a strange case and took so long to track down exactly under what conditions it occurs, that I wanted to pass it along in case it crops up again or is worth fixing.
##### Steps To Reproduce:

Create an inventory:

```
[testservers]
localhost

[testservers:var]
foo = foo
subdir = subdir
```

In `subdir/vars.yml` I have:

```

---
quux: quux
```

And a play in `test.yml`:

```

---
- hosts: testservers
  connection: local
  vars_files:
    - "{{ subdir }}/vars.yml"
  roles:
    - role: test
    - role: test
      foo: bar
```

The `test` role simply runs the task:

```

---
# roles/test/tasks/main.yml
- debug: var=foo
```
##### Expected Results:

```
GATHERING FACTS ***************************************************************
ok: [localhost]
localhost: importing /Users/nate/ansible/test/subdir/vars.yml

TASK: [test | debug var=foo] **************************************************
ok: [localhost] => {
    "foo": "foo"
}

TASK: [test | debug var=foo] **************************************************
ok: [localhost] => {
    "foo": "bar"
}
```
##### Actual Results:

```
GATHERING FACTS ***************************************************************
ok: [localhost]
localhost: importing /Users/nate/ansible/test/subdir/vars.yml

TASK: [test | debug var=foo] **************************************************
ok: [localhost] => {
    "foo": "foo"
}

TASK: [test | debug var=foo] **************************************************
ok: [localhost] => {
    "foo": "foo"
}
```

**Comment by natefoo** [comment]
Whoops. Updated original description using the template.

**Comment by abadger** [comment]
@natefoo Just tested this and it looks like it was fixed by 8a1fbed5d6a61583950024df60ac74d5c98d6cca

**Comment by abadger** [comment]
Added an integration test to check this case: a9b5b1bf55728b5d9dd38dc614be491a2c4217f0 and closing.  Thanks for reporting it!
